eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -95,7 +95,7 @@ object Helpers {
     if (open.pushMsat > open.fundingSatoshis) return Left(InvalidPushAmount(open.temporaryChannelId, open.pushMsat, open.fundingSatoshis.toMilliSatoshi))
 
     // BOLT #2: The receiving node MUST fail the channel if: to_self_delay is unreasonably large.
-    if (open.toSelfDelay > Channel.MAX_TO_SELF_DELAY || open.toSelfDelay > nodeParams.channelConf.maxToLocalDelay) return Left(ToSelfDelayTooHigh(open.temporaryChannelId, open.toSelfDelay, nodeParams.channelConf.maxToLocalDelay))
+    if (open.toSelfDelay > nodeParams.channelConf.maxToLocalDelay) return Left(ToSelfDelayTooHigh(open.temporaryChannelId, open.toSelfDelay, nodeParams.channelConf.maxToLocalDelay))
```

Comment:
To be clear, the previous check was redundant right?

Reply:
It was redundant when `nodeParams.channelConf.maxToLocalDelay` was <= 2016 (default from `reference.conf`).

If the user had set `nodeParams.channelConf.maxToLocalDelay` to something greater than 2016, eclair would still reject channels with a `to_self_delay` greater than 2016, which was confusing for users, but now it won't reject them.